src/Pim/Component/User/ReadModel/AuthenticatedUser.php

```
@@ -0,0 +1,201 @@
+
+    /** @var array */
+    private $roles;
+
+    /** @var bool */
+    private $enabled;
+
+    /** @var string */
+    private $salt;
+
+    /** @var string */
+    private $uiLocale;
+    /** @var string */
+    private $email;
+
+    /**
+     * @param int    $id
+     * @param string $username
+     * @param string $password
+     * @param array  $roles
+     * @param bool   $enabled
+     * @param string $salt
+     * @param string $uiLocale
+     * @param string $email
+     */
+    public function __construct(
+        int $id,
+        string $username,
+        string $password,
+        array $roles,
+        bool $enabled,
+        string $salt,
+        string $uiLocale,
+        string $email
+    ) {
+        $this->id = $id;
+        $this->username = $username;
+        $this->password = $password;
+        $this->roles = $roles;
+        $this->enabled = $enabled;
+        $this->salt = $salt;
+        $this->uiLocale = $uiLocale;
+        $this->email = $email;
+    }
+
+    /**
+     * @return int
+     */
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return string
+     */
+    public function getUsername(): string
+    {
+        return $this->username;
+    }
+
+    /**
+     * @return string
+     */
+    public function getPassword(): string
+    {
+        return $this->password;
+    }
+
+    /**
+     * @return array
+     */
+    public function getRoles(): array
+    {
+        return array_map(
+            function ($role) {
+                return $role['role'];
+            },
+            $this->roles
+        );
+
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSalt(): string
+    {
+        return $this->salt;
+    }
+
+    /**
+     * @return string
+     */
+    public function getUiLocale(): string
+    {
+        return $this->uiLocale;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function eraseCredentials(): void
+    {
```

maybe we could throw an exception here as it's not normal that this method is called on a read only object

---

dunno this method is used to reset plain password. We don't need it here but I am not sure that is a good idea to raise an exception.

---

This object is readonly, so it means that we should not implement method that should transform the object. Here, the method name suggests that it should do something. It comes from the interface so you have to implement it but we should not use it. That's why I suggest to throw error

---

You don't change the state of this object and I don't know when the security component calls this method. Moreover the added value of object is to to decouple Symfony security component from our write model.